platform/osgi/platform-osgi-configadmin/src/test/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManagerTest.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.felix.cm.PersistenceManager;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EncryptingWrappedPersistenceManagerTest {
+    private static final String PID = "my_pid";
+
+    private static final String VALUE_TO_ENCRYPT = "some_string";
+
+    private static final Dictionary<String, Object> TEST_CASES = new Hashtable<>();
+
+    static {
+        // We don't encrypt integers
+        TEST_CASES.put("1", 3);
+        // We SHOULD encrypt strings
+        TEST_CASES.put("2", VALUE_TO_ENCRYPT);
+        // Except the following three strings which are special for OSGi / Felix
+        TEST_CASES.put(SERVICE_PID, PID);
+        TEST_CASES.put(SERVICE_FACTORYPID, PID);
+        TEST_CASES.put(FELIX_FILENAME, PID);
+    }
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Mock
+    private PersistenceManager mockManager;
+
+    private EncryptingWrappedPersistenceManager persistenceManager;
+
+    @Before
+    public void setup() throws Exception {
+        String pwDirectory = temporaryFolder.newFolder()
+                .getAbsolutePath();
+        persistenceManager = new EncryptingWrappedPersistenceManager(mockManager, pwDirectory);
+    }
+
+    @Test
+    public void testRoundTrip() throws Exception {
+        ArgumentCaptor<Dictionary> captor = ArgumentCaptor.forClass(Dictionary.class);
+        persistenceManager.store(PID, TEST_CASES);
+        verify(mockManager).store(eq(PID), captor.capture());
+
+        Dictionary<String, Object> capturedProps = captor.getValue();
+        String valueToEncrypt = (String) capturedProps.get("2");
+        assertThat(capturedProps.size(), is(5));
+        assertSpecialStrings(capturedProps);
+
+        assertThat("VALUE_TO_ENCRYPT was not actually encrypted",
+                valueToEncrypt,
+                is(not(VALUE_TO_ENCRYPT)));
+        assertThat("Encrypting VALUE_TO_ENCRYPT results in a super-sequence of the original",
+                valueToEncrypt,
+                not(containsString(VALUE_TO_ENCRYPT)));
+        assertThat("VALUE_TO_ENCRYPT has an encrypted value that is too short",
+                valueToEncrypt.length(),
+                is(greaterThan(VALUE_TO_ENCRYPT.length() * 2)));
+
+        when(mockManager.load(PID)).thenReturn(capturedProps);
+
+        Dictionary<String, Object> loadedProps = persistenceManager.load(PID);
+        assertThat(loadedProps.size(), is(5));
+        assertSpecialStrings(loadedProps);
+
+        assertThat("VALUE_TO_ENCRYPT should be its original value prior to encryption",
+                loadedProps.get("2"),
+                is(VALUE_TO_ENCRYPT));
+    }
+
+    private static void assertSpecialStrings(Dictionary<String, Object> props) {
```

Comment:
✏️  Had to go to this method from the test above in order to understand what it did. A more explicit name like `assertInternalStringsDidNotChange()` would help.